src/node-jspm-jasmine.js

```diff
@@ -141,17 +179,79 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// jasmine will exit the process on its own, and since there doesn't appear
+		// to be a "onExit" hook, we're forced to intercept node's exit function and
+		// use as a "all done" jasmine hook.
+		// jasmine does offer a "addReporter" option, which contains a "jasmineDone"
+		// hook, however, it has no mechanism for waiting on async operations and
+		// async is needed for importing modules.
+		// Also, we should provide a "onExit" hook for the "programmatic" version,
+		// otherwise jasmine will abruptly kill the current running script.
+		const processExit = process.exit;
+		process.exit = function () {
```

Comment:
I'm thinking about what we can do for this, and have two potential ideas:

1) Instrument all the files *before* jasmine, so that we can stick with the synchronous jasmineDone hook. Let me know if there's some reason why this would mess with the coverage results.
2) Still use the jasmineDone hook, but add a `process.on('exit')` event handler *that only prevents exiting the program exactly one time* (because we know that jasmine is about to try to exit). That way the user can still kill the program, but we can hook into when jasmine is done.

I like 1) a lot better than 2), if it's possible. What are your thoughts?